lua/ulx/modules/sh/cfc_checklua.lua

```diff
@@ -2,40 +2,114 @@ CFCUlxCommands.checklua = CFCUlxCommands.checklua or {}
 
 if SERVER then
-    util.AddNetworkString( "CFC_ULX_StatCheckCL" )
-    util.AddNetworkString( "CFC_ULX_StatCheckSV" )
-end
+    util.AddNetworkString( "CFC_ULX_StatCheck" )
+    util.AddNetworkString( "CFC_ULX_AutoStatCheck" )
+
+    local awaitingResponse = {}
+
+    local function sendAwarn( ply )
+        if not awarn_warnplayer then return end
+
+        awarn_warnplayer( nil, ply, AUTO_BAN_REASON )
+    end
+
+    local function receiveConvars( ply )
+        local any = false
+        local badConvars = {}
+
+        for _, convar in pairs( CONVARS_TO_CHECK ) do
+            local convarValue = net.ReadBool()
+
+            if convarValue then
+                any = true
+                badCovnars[convar] = convarValue
+                ulx.fancyLogAdmin( awaitingResponse[ply], true, "#T's " .. convar .. "value is " .. tostring( convarValue ), ply )
+            end
+        end
+
+        return any, badConvars
+    end
+
+    local function unawaitTimerName( steamId )
+        return "CFC_ULX_AutoStatsUnAwait_" .. steamId
+    end
+
+    net.Receive( "CFC_ULX_StatCheck", function( _, ply )
+        if not awaitingResponse[ply] then return end
+        awaitingResponse[ply] = nil
+
+        receiveConvars( ply )
+    end )
 
-local awaitingResponse = {}
+    net.Receive( "CFC_ULX_AutoStatCheck", function( _, ply )
+        if not awaitingResponse[ply] then return end
+        awaitingResponse[ply] = nil
+        timer.Remove( unawaitTimerName( ply:GetSteamID() ) )
 
-net.Receive( "CFC_ULX_StatCheckSV", function( _, ply )
-    if not awaitingResponse[ply] then return end
-    local convar = net.ReadBool()
-    ulx.fancyLogAdmin( awaitingResponse[ply], true, "#T's sv_allowcslua value is " .. tostring( convar ), ply )
+        local anyConvars, badConvars = receiveConvars( ply )
 
-    awaitingResponse[ply] = nil
-end )
+        if anyConvars then
+            print( "Banning " .. tostring( ply ) .. " for having bad convars." )
+            PrintTable( badConvars )
 
-function cmd.checkluaPlayers( callingPlayer, targetPlayers )
-    for _, ply in pairs( targetPlayers ) do
-        awaitingResponse[ply] = callingPlayer
-        net.Start( "CFC_ULX_StatCheckCL" )
-        net.Send( ply )
+            ulx.ban( nil, ply, AUTO_BAN_LENGTH, AUTO_BAN_REASON )
+            sendAwarn( ply )
+        end
+    end )
+
+    function cmd.checkluaPlayers( callingPlayer, targetPlayers )
+        for _, ply in pairs( targetPlayers ) do
+            awaitingResponse[ply] = callingPlayer
+            net.Start( "CFC_ULX_StatCheckCL" )
+            net.Send( ply )
+        end
     end
+
+    gameevent.Listen( "player_connect" )
+    hook.Add( "player_connect", "CFC_ULX_AutoStatsCheckOnConnect", function( data )
+        local steamId = data.networkid
+
+        local ply = player.GetBySteamID( steamId )
+        awaitingResponse[ply] = true
+
+        timer.Create( unawaitTimerName( steamId ), 60 * 15, 1, function()
```

Comment:
We wait 15 minutes for a reply?
Seems a little long
Also is it possible that ply is still null by the time this hook is called?

Reply:
This is just for loading

Reply:
Wouldn't the InitPostEntity hook at line 101 handle that?

Reply:
Unless they don't finish loading

Reply:
If they don't finish loading is there even a point in checking the convars?

Reply:
Nope, which is why we stop expecting a response from them in that case

Reply:
This chunk of code says "When they join, I'll wait for 15 minutes to hear back about their convars, otherwise assume they didn't finish loading"

Reply:
Yeah but like why'd it even need to know that the player has connected? wouldn't a InitPostEntity hook just do the job once the player has actually connected?

Reply:
That's on the clientside, the idea here is that we're expecting them to respond so that we can ignore any unexpected network messages.

Reply:
the InitPostEntity hook is server side

Reply:
The one on `102 `is clientside

Reply:
wouldn't running a InitPostEntity server hook then sending a net message to the player and one back be the most efficient and fool proof way of checking it on join?

Reply:
`InitPostEntity` will run one time on the server after it starts up, so that's not the right one here.
If you mean `PlayerInitialSpawn`, that could work, but the client isn't ready to receive net messages at that point.

We have a `PlayerFullLoad` lib that would work here, but I didn't want to make that a dependency of our ULX commands, and I didn't want to use a timer in `PlayerInitialSpawn` for it either.

Reply:
oh fuck yeah mb you're right i confused it, sorry.